iohub/convert.py

```diff
@@ -307,15 +307,15 @@ def _get_pos_names(self):
         """Append a list of pos names in ascending order
         (order in which they were acquired).
         """
-        if self.p > 1:
+        try:
             self.pos_names = []
             for p in range(self.p):
                 try:
                     name = self.reader.stage_positions[p]["Label"]
                 except (IndexError, KeyError):
                     name = p
                 self.pos_names.append(name)
-        else:
+        except Exception:
```

Comment:
Can you enumerate the expected exceptions here?

Reply:
Also what happens if the dataset has multiple positions but gets caught in this branch? Should it `raise`?

Reply:
The confusing part seems to be the nested `try` statements, which are not necessary. Simplified.